pkg/scheduler/framework/v1alpha1/interface.go

```diff
@@ -139,4 +175,13 @@ type FrameworkHandle interface {
 	// a pod finishes "Reserve" point. There is no guarantee that the information
 	// remains unchanged in the binding phase of scheduling.
 	NodeInfoSnapshot() *internalcache.NodeInfoSnapshot
+
+	// IterateOverWaitingPods acquires a read lock and iterates over the WaitingPods map.
+	IterateOverWaitingPods(callback func(types.UID, WaitingPod))
```

Comment:
Why does `callback` take a UID? UID can be retrieved from the pod object if needed.

Reply:
Done.